src/Database/Adapter/MariaDB.php

```diff
@@ -246,35 +245,222 @@ public function updateAttribute(string $collection, string $id, string $type, in
      * @throws Exception
      * @throws PDOException
      */
-    public function renameIndex(string $collection, string $old, string $new): bool
+    public function renameAttribute(string $collection, string $old, string $new): bool
     {
         $collection = $this->filter($collection);
         $old = $this->filter($old);
         $new = $this->filter($new);
 
         return $this->getPDO()
-            ->prepare("ALTER TABLE {$this->getSQLTable($collection)} RENAME INDEX `{$old}` TO `{$new}`;")
+            ->prepare("ALTER TABLE {$this->getSQLTable($collection)} RENAME COLUMN `{$old}` TO `{$new}`;")
             ->execute();
     }
 
     /**
-     * Delete Attribute
-     *
      * @param string $collection
      * @param string $id
-     * @param bool $array
+     * @param string $type
+     * @param string $relatedCollection
+     * @param bool $twoWay
+     * @param string $twoWayKey
+     * @param string $onDelete
      * @return bool
      * @throws Exception
-     * @throws PDOException
      */
-    public function deleteAttribute(string $collection, string $id, bool $array = false): bool
-    {
+    public function createRelationship(
+        string $collection,
+        string $relatedCollection,
+        string $type,
+        bool $twoWay = false,
+        string $id = '',
+        string $twoWayKey = '',
+        string $onDelete = Database::RELATION_MUTATE_RESTRICT
+    ): bool {
         $name = $this->filter($collection);
+        $relatedName = $this->filter($relatedCollection);
+        $table = $this->getSQLTable($name);
+        $relatedTable = $this->getSQLTable($relatedName);
         $id = $this->filter($id);
+        $sqlType = $this->getSQLType(Database::VAR_RELATIONSHIP, 0, false);
+
+        switch ($type) {
+            case Database::RELATION_ONE_TO_ONE:
+                $sql = "ALTER TABLE {$table} ADD COLUMN `{$id}` {$sqlType} DEFAULT NULL;";
+
+                if ($twoWay) {
+                    $sql .= "ALTER TABLE {$relatedTable} ADD COLUMN `{$twoWayKey}` {$sqlType} DEFAULT NULL;";
+                }
+                break;
+            case Database::RELATION_ONE_TO_MANY:
+                $sql = "ALTER TABLE {$relatedTable} ADD COLUMN `{$twoWayKey}` {$sqlType} DEFAULT NULL;";
+                break;
+            case Database::RELATION_MANY_TO_ONE:
+                $sql = "ALTER TABLE {$table} ADD COLUMN `{$id}` {$sqlType} DEFAULT NULL;";
+                break;
+            case Database::RELATION_MANY_TO_MANY:
+                return true;
+            default:
+                throw new Exception('Invalid relationship type.');
+        }
```

Comment:
```suggestion
        $sql = '';
        switch ($type) {
            case Database::RELATION_ONE_TO_ONE:
                $sql = "ALTER TABLE {$table} ADD COLUMN `{$id}` {$sqlType} DEFAULT NULL;";

                if ($twoWay) {
                    $sql .= "ALTER TABLE {$relatedTable} ADD COLUMN `{$twoWayKey}` {$sqlType} DEFAULT NULL;";
                }
                break;
            case Database::RELATION_ONE_TO_MANY:
                $sql = "ALTER TABLE {$relatedTable} ADD COLUMN `{$twoWayKey}` {$sqlType} DEFAULT NULL;";
                break;
            case Database::RELATION_MANY_TO_ONE:
                $sql = "ALTER TABLE {$table} ADD COLUMN `{$id}` {$sqlType} DEFAULT NULL;";
                break;
            case Database::RELATION_MANY_TO_MANY:
                break;
            default:
                throw new Exception('Invalid relationship type.');
        }
        
        if (empty($sql)) {
            return true;
        }
```

to be consistent with the `updateAttribute` method below, we can declare $sql to be an empty string outside the switch case and return `true` outside the switch case if its still empty

Reply:
We need to do that for the update and delete because there's a possibility it won't be assigned, but that's not the case here